--- wine/dlls/win32u/driver.c.orig	2023-04-02 22:26:28
+++ wine/dlls/win32u/driver.c	2024-02-10 03:19:48
@@ -950,7 +950,9 @@
     static const WCHAR prop_nameW[] =
         {'_','_','w','i','n','e','_','d','i','s','p','l','a','y','_','d','e','v','i','c','e',
          '_','g','u','i','d',0};
-
+    static const WCHAR winemac_drv[] =
+        {'w','i','n','e','m','a','c','.','d','r','v',0};
+  
     user_check_not_lock();
 
     asciiz_to_unicode( driver_load_error, "The explorer process failed to start." );  /* default error */
@@ -960,6 +962,7 @@
     guid_atom = HandleToULong( NtUserGetProp( hwnd, prop_nameW ));
     memcpy( key, guid_key_prefixW, sizeof(guid_key_prefixW) );
     ptr = key + ARRAYSIZE(guid_key_prefixW);
+#if 0
     if (NtQueryInformationAtom( guid_atom, AtomBasicInformation, buf, sizeof(buf), NULL ))
         return FALSE;
     memcpy( ptr, abi->Name, abi->NameLength );
@@ -968,10 +971,13 @@
     ptr += ARRAY_SIZE(guid_key_suffixW);
 
     if (!(hkey = reg_open_key( NULL, key, (ptr - key) * sizeof(WCHAR) ))) return FALSE;
+#endif
 
-    if ((size = query_reg_ascii_value( hkey, "GraphicsDriver", info, sizeof(buf) )))
+    if (TRUE /*(size = query_reg_ascii_value( hkey, "GraphicsDriver", info, sizeof(buf) ))*/)
     {
         static const WCHAR nullW[] = {'n','u','l','l',0};
+        memcpy(info->Data, winemac_drv, 12 * sizeof(WCHAR));
+        info->DataLength = 12;
         TRACE( "trying driver %s\n", debugstr_wn( (const WCHAR *)info->Data,
                                                   info->DataLength / sizeof(WCHAR) ));
         if (info->DataLength != sizeof(nullW) || memcmp( info->Data, nullW, sizeof(nullW) ))
